org.uqbar.project.wollok.lib/src/wollok.wlk

```diff
@@ -493,6 +493,91 @@ package lang {
 	}
 	
 	/**
+	 * Represents a set of key -> values
+	 * 
+	 */
+	class Dictionary {
+	
+		constructor() { }
+		/**
+		 * Adds or updates a value based on a key
+		 */
+		method put(_key, _value) native
+		
+		/*
+		 * Returns the value to which the specified key is mapped, or null if this Dictionary contains no mapping for the key.
```

Comment:
Fer already changed it, although he may not have pushed it yet. Now there
is a basicGet (like this one) a standard get, which throws an exception
when the key is not found, and a getOrElse that lets you choose the
alternative using a block.

On Tue, Jun 21, 2016 at 6:10 PM, Nico Passerini <notifications@github.com>
wrote:

> In org.uqbar.project.wollok.lib/src/wollok.wlk
> <https://github.com/uqbar-project/wollok/pull/753#discussion_r67952397>:
>
> > @@ -493,6 +493,91 @@ package lang {
> >  	}
> >  	
> >  	/**
> > +	 * Represents a set of key -> values
> > +	 *
> > +	 */
> > +	class Dictionary {
> > +	
> > +		constructor() { }
> > +		/**
> > +		 * Adds or updates a value based on a key
> > +		 */
> > +		method put(_key, _value) native
> > +		
> > +		/*
> > +		 * Returns the value to which the specified key is mapped, or null if this Dictionary contains no mapping for the key.
>
> This implies that the dictionary can not differentiate not having a key
> from associating a null value with that key. I do not say it is wrong, just
> bring it up to be sure that it is what we mean.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/wollok/pull/753/files/60ac511651e02d8ad50609b0d50d3ce6af005d94#r67952397>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ-EZtYC9sz5pZ3q-ZMcBVhwtTboBUxwks5qOFNCgaJpZM4I5Y_O>
> .
>